src/core/WarpFactory.ts

```diff
@@ -0,0 +1,136 @@
+import Arweave from 'arweave';
+import { CacheableExecutorFactory, Evolve } from '@warp/plugins';
+import {
+  CacheableStateEvaluator,
+  ConfirmationStatus,
+  EvalStateResult,
+  HandlerExecutorFactory,
+  WARP_GW_URL,
+  SourceType,
+  Warp,
+  WarpBuilder
+} from '@warp/core';
+import { LevelDbCache, MemCache } from '@warp/cache';
+
+export type GatewayOptions = {
+  confirmationStatus: ConfirmationStatus;
+  source: SourceType;
+  address: string;
+};
+
+export type CacheOptions = {
+  maxStoredTransactions: number;
+  inMemory: boolean;
+  dbLocation?: string;
+};
+
+export const defaultWarpGwOptions: GatewayOptions = {
+  confirmationStatus: { notCorrupted: true },
+  source: null,
+  address: WARP_GW_URL
+};
+
+export const defaultCacheOptions: CacheOptions = {
+  maxStoredTransactions: 10,
+  inMemory: false
+};
+
+/**
+ * A factory that simplifies the process of creating different versions of {@link Warp}.
+ * All versions use the {@link Evolve} plugin.
+ */
+export class WarpFactory {
+  /**
+   * creates a Warp instance suitable for testing in a local environment
+   * (e.g. usually using ArLocal)
+   * @param arweave - an instance of Arweave
+   * @param cacheOptions - optional cache options. By default, the in-memory cache is used.
+   */
+  static forLocal(
```

Comment:
WDYT about making arweave an optional field and creating an instance if it's not passed to the factory?